CONTRIBUTING.md

```diff
@@ -1,135 +1,387 @@
+
+---
+
+## Contributing Changes
+
+### Workflow
+
+1. **Open a GitHub issue** ([HDF5 Issues](https://github.com/HDFGroup/hdf5/issues))
+   - **Required** unless the change is minor (e.g., typo fix).
+   - Describe the problem or feature request clearly.
+
+2. **Fork the repository** and create your branch
+   - Target the `develop` branch for new features and bug fixes.
+   - Use descriptive branch names.
+
+3. **Make your changes**
+   - Follow HDF5 coding conventions.
+   - Add tests for new functionality or bug fixes.
+   - Update documentation as needed.
+
+4. **Build and test thoroughly**
+   - Follow build instructions in `release_docs/INSTALL*` files.
+   - Ensure all tests pass.
+
+5. **Submit a pull request (PR)**
+   - Address any formatting or testing issues reported by CI.
+   - Make sure to include the issue that the PR addresses in the description.
+   - Work with HDF Group developers to meet acceptance criteria.
+
+### Acceptance Criteria
+
+For a pull request to be accepted, it must satisfy:
+
+* **Clear purpose:** What does it address? How does it benefit the HDF5 community?
+* **Proper documentation:** Code must be documented for maintainability.
+* **Testing:** Must pass HDF5 regression testing and include appropriate tests.
+  - We do not expect you to perform comprehensive testing across multiple platforms
+    before we accept the pull request.
+* **Compatibility:** Must not compromise HDF5's core principles:
+  - 100% backward compatibility (any HDF5 file must remain readable).
+    - If your patch's purpose is to modify the HDF5 data model or file format, **please** discuss
+      this with us first. File format changes and features required by those changes can be
+      introduced only in a new major release.
+  - Machine independence (data readable across all platforms).
+  - Binary compatibility for maintenance releases (no changes to public APIs/structures).
+* **Documentation:** New features must be properly documented. This includes using Doxygen
+    and providing information in release documents such as `RELEASE.txt`.
+
+### Branching Strategy
+
+* **Small features:** Develop in forks of the main repository.
+* **Large collaborative work:** Use feature branches named `feature/<feature>` in the main repository.
+* Add `BRANCH.md` file explaining branch purpose and contact info for feature branches.
+
+---
+
+## Testing
+
+### Test Structure
+
+HDF5 uses custom testing macros rather than standard frameworks. There are two systems:
+
+#### Modern Testing (`h5test.h`) - Preferred
+```c
+#include "h5test.h"
 
+static int
+test_feature(void)
+{
+    TESTING("some feature");
+    
+    /* test code */
+    if (error_condition)
+        TEST_ERROR;
+    
+    PASSED();
+    return SUCCEED;
+
+error:
+    return FAIL;
+}
+```
+
+#### Legacy Testing (`testhdf5.h`) - Avoid for New Code
+Used only by the large `testhdf5` program. Uses global variables and should be avoided.
```

Comment:
I know using global variables should be avoided, but I feel that "Uses global variables and should be avoided" is a bit unclear.  Does it say using global variables should be avoided or legacy testing uses global variables and should be avoided?  If it's my English, you can ignore this comment. :D